.github/workflows/win.yml

```
@@ -138,6 +144,41 @@ jobs:
           mv libyaml/Release/yaml.lib libs/
           mv libxml2/Release/libxml2s.lib libs/xml2.lib
 
+      - name: Cache OpenSSL
+        id: cache-openssl
+        uses: actions/cache@v2
+        with:
+          path: |
+            libs/crypto.lib
+            libs/ssl.lib
+          key: win-openssl-libs-3.0.0
+      - name: Set up NASM
+        if: steps.cache-openssl.outputs.cache-hit != 'true'
+        uses: ilammy/setup-nasm@v1.2.1
```

Please pin the commit for security
```suggestion
        uses: ilammy/setup-nasm@e2335e5fc95548c09cd2deea2768793e0e8f0941 # v1.2.1
```